pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -18,33 +18,101 @@ package machinesync
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
+	configv1 "github.com/openshift/api/config/v1"
 	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/cluster-capi-operator/pkg/util"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	awscapiv1beta1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta1"
+	capiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	capiNamespace string = "openshift-cluster-api"
+	mapiNamespace string = "openshift-machine-api"
+)
+
+var (
+	// errPlatformNotSupported is returned when the platform is not supported.
+	errPlatformNotSupported = errors.New(`error determining InfrastructureMachine
+	type, platform not supported`)
+)
```

Comment:
It looks like the CAPI contract and types calls these infrastructure specific machines: `InfraMachine`(s) (https://github.com/search?q=repo%3Akubernetes-sigs%2Fcluster-api%20InfraMachine&type=code)

I'd probably rename the InfrastructureMachine and Infrastructure Machine occurrences to InfraMachine :)